### PR TITLE
Fix current Kubernetes namespace parsing issues

### PR DESCRIPTION
**Pull request by captjt**
Added functionality to parse the current namespace regardless of config settings. This should address the issue [#555](https://github.com/bhilburn/powerlevel9k/issues/555).

### CODE REVIEW

File: powerlevel9k.zsh-theme [hunk 1]
```diff
@@ -1410,20 +1410,22 @@ prompt_dir_writable() {
...
-# Kubernetes Current Context
+# Kubernetes Current Context/Namespace
 prompt_kubecontext() {
   local kubectl_version="$(kubectl version --client 2>/dev/null)"
 
   if [[ -n "$kubectl_version" ]]; then
-    # Get the current Kubernetes config context's namespaece
-    local k8s_namespace=$(kubectl config get-contexts --no-headers | grep '*' | awk '{print $5}')
+    local cur_ctx=$(kubectl config view -o=jsonpath='{.current-context}')
+    cur_namespace="$(kubectl config view -o=jsonpath="{.contexts[?(@.name==\"${cur_ctx}\")].context.namespace}")"
```
Comment (derekbassett) [inline powerlevel9k.zsh-theme]: Do you know what version of Kubectl they added support for -o=jsonpath?

Comment (captjt) [reply powerlevel9k.zsh-theme]: Hmmm.. The earliest official version on kubernetes.io is v1.5 and it supports `jsonpath` flag --> [kubectl config view](https://kubernetes.io/docs/user-guide/kubectl/v1.5/#-em-view-em-).

**Comment by bhilburn** [comment]
@captjt - Thanks so much for submitting this, and thanks to @derekbassett for providing testing and review! It looks like the unit tests are failing with this:

```
ASSERT:expected:
    <%K{magenta} %F{white%}⎈%f %F{white}minikube/default %k%F{magenta}%f > 
but was:
    <%K{magenta} %F{white%}⎈%f %F{white}/default %k%F{magenta}%f >

ASSERT:expected:
    <%K{magenta} %F{white%}⎈%f %F{white}minikube/kube-system %k%F{magenta}%f > 
but was:
    <%K{magenta} %F{white%}⎈%f %F{white}/default %k%F{magenta}%f >
```

So it's failing because the `minikube` string disappeared. I'm not a kubernetes user (indeed, I know next to nothing about containers), but my guess, here, is that that's basically a username or namespace and `default` is some sort of instance? Regardless, is there value in having that there, and is it easy to fix?

**Comment by captjt** [comment]
I can check this out. I don't remember writing these unit tests maybe they were added after we merged in the first take on kube context's feature. I'll take a look as soon as I can get the time. Thanks @bhilburn!!

File: powerlevel9k.zsh-theme (hunk 1 above)
Comment (derekbassett) [reply powerlevel9k.zsh-theme]: Thought it might be important to highlight that.

**Comment by derekbassett** [comment]
Hey so I figured out what was going on.  The mock was not correctly updated.  Also @captjt it looks like you are calling kubectl three time rather then just two times.  I've got the fix in my repo https://github.com/derekbassett/powerlevel9k/tree/fix/kubernetes-prompt.  I thought I was able to PR it to your branch @captjt but my github foo is failing me.

**Comment by captjt** [comment]
Going to close this already approved @derekbassett's PR [#661](https://github.com/bhilburn/powerlevel9k/pull/661). Thanks for taking the time to fix the tests and clean up the code! 😄 

cc. @bhilburn

**Comment by derekbassett** [comment]
Your quite welcome! I wouldn’t have thought to look for config context view. Glad that we got it figured out.

**Pull request closed without merging**